homeassistant/components/modem_callerid/sensor.py

```diff
@@ -28,33 +25,47 @@
     }
 )
 
+SERVICE_SCHEMA = vol.Schema({vol.Optional(ATTR_ENTITY_ID): cv.comp_entity_ids})
+
+_LOGGER = logging.getLogger(__name__)
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up modem caller ID sensor platform."""
 
-    name = config.get(CONF_NAME)
-    port = config.get(CONF_DEVICE)
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Modem Caller ID sensor."""
+    name = entry.data[CONF_NAME]
+    device = entry.data[CONF_DEVICE]
+    api = hass.data[DOMAIN][entry.entry_id][DATA_KEY_API]
+    sensor = [
+        ModemCalleridSensor(
+            hass,
+            api,
+            name,
+            device,
+            entry.entry_id,
+        )
+    ]
+    async_add_entities(sensor, True)
 
-    modem = bm(port)
-    if modem.state == modem.STATE_FAILED:
-        _LOGGER.error("Unable to initialize modem")
-        return
+    platform = entity_platform.current_platform.get()
 
-    add_entities([ModemCalleridSensor(hass, name, port, modem)])
+    platform.async_register_entity_service(
+        SERVICE_REJECT_CALL, SERVICE_SCHEMA, "reject_call"
```

Comment:
```suggestion
        SERVICE_REJECT_CALL, {}, "reject_call"
```

I think you can use an empty dict here to have the entity field auto added since its an entity service

Reply:
Testing leaves me with no target options. I think we have to leave this in.